---
title: "E013 drawing CgPho4-ScPho4 alignment"
author: "Bin He"
date: "2024-03-01 updated `r Sys.Date()`"
output:
  html_notebook:
    theme: flatly
    toc: yes
    code_folding: hide
---

```{r message=FALSE}
require(tidyverse)
require(cowplot)
require(seqinr)
require(ggnewscale)
#require(ggtext)
#require(RColorBrewer)
```

```{r}
old <- theme_set(theme_bw(base_size = 16))
```

# Goal
Create illustrative figures to visualize the level of similarity between ScPho4 and CgPho4, as well as annotations.

# Process alignment
The idea is to construct a table from the MSA so we can map the positions within each Pho4 ortholog to a common coordinate based on the alignment. We will incorporate alignment scores calcucated by Jalview as additional columns. Later, we can map other external annotation scores, such as PADDLE prediction, onto the same coordinate system.

## Read alignment
Using the `seqinr` package
```{r}
msa <- read.alignment(
  file = "../input/Pho4-alignment/20201124-ScPho4-CgPho4-aa-aln.fa", 
  format = "fasta")
tmp <- str_split(msa$seq, pattern = "", simplify = FALSE)
names(tmp) <- msa$nam
aln <- tibble(
  pos_aln = 1:length(tmp$Scer),
  char_Scer = toupper(tmp$Scer),
  pos_Scer = cumsum(char_Scer != "-"),
  char_Cgla = toupper(tmp$Cgla),
  pos_Cgla = cumsum(char_Cgla != "-")
)
```

## Read Jalview score
I exported the autocalculated scores by Jalview. We will be using the 
```{r}
jalview.score <- read.csv(
  "../input/Pho4-alignment/20240301-ScPho4-CgPho4-jalview-score.csv",
  header = FALSE, row.names = 1
  ) %>% t() %>% # transpose the data matrix
  as_tibble()
# check to make sure that this table has the same length as the alignment table
# if so, we assume the rows correspond to the same positions
if(nrow(aln) == nrow(jalview.score)){
  dat <- cbind(aln, jalview.score)
}
```

## Calculate the average score per region
Encode the breakpoints
```{r}
r.label <- paste0("R", 1:5)
br = list(
  Scer = c(0, 42, 99, 176, 242, 312),
  Cgla = c(0, 44, 112, 282, 458, 533),
  aln = c(0, 44, 113, 294, 470, 547)
)
# make a tibble containing the region definition
regions <- tibble(
  name = r.label,
  begin_Scer = head(br$Scer + 1, -1),
  end_Scer = tail(br$Scer, -1),
  begin_Cgla = head(br$Cgla + 1, -1),
  end_Cgla = tail(br$Cgla, -1),
  begin_aln = head(br$aln + 1, -1),
  end_aln = tail(br$aln, -1)
)

```

Calculate the average scores
```{r}
dat %>% mutate(region = cut(pos_Scer, breaks = br$Scer, labels = r.label)) %>% 
  group_by(region) %>% 
  summarize(across(c(Conservation, Quality), mean))
```
## Plot alignment line graph
```{r}
lw = 0.6
p <- dat %>% 
  mutate(
    pos_aln = pos_aln - 1/2*max(pos_aln),
    pos_Scer = ifelse(char_Scer == "-", NA, pos_Scer - 1/2*max(pos_Scer)),
    pos_Cgla = ifelse(char_Cgla == "-", NA, pos_Cgla - 1/2*max(pos_Cgla))
  ) %>% 
  ggplot() +
  geom_tile(aes(x = pos_aln, y = 1.5, height = 0.1, fill = Conservation), 
            width = 1.5, linewidth = 0) +
  geom_segment(aes(x = pos_aln, xend = pos_Scer, y = 1.55, yend = 1.95), color = "gray90") +
  geom_segment(aes(x = pos_Cgla, xend = pos_aln, y = 1.05, yend = 1.45), color = "gray90") +
  geom_rect(data = regions,
            aes(xmin = begin_Scer - 312/2, xmax = end_Scer - 312/2,
                ymin = 1.95, ymax = 2.15), 
            fill = "forestgreen", color = "gray90", linewidth = lw) +
  geom_rect(data = regions,
            aes(xmin = begin_Cgla - 533/2, xmax = end_Cgla - 533/2,
                ymin = 0.85, ymax = 1.05), 
            fill = "steelblue2", color = "gray90", linewidth = lw) +
  geom_segment(data = as_tibble(br),
               aes(x = aln - 547/2, xend = Scer - 312/2,  y = 1.55, yend = 1.95),
               color = "gray40", linewidth = lw, linetype = 3) +
  geom_segment(data = as_tibble(br),
               aes(xend = aln - 547/2, x = Cgla - 532/2,  y = 1.05, yend = 1.45),
               color = "gray40", linewidth = lw, linetype = 3) +
  #scale_y_continuous(expand = expansion(mult = 0.3)) +
  scale_fill_distiller(palette = "Greys", direction = 1, breaks = c(-0.1,5,10,12))
p +
  theme_cowplot() +
  theme(legend.position = "none",
        axis.line = element_blank(),
        axis.text = element_blank(),
        axis.title = element_blank(),
        axis.ticks = element_blank())
```
```{r}
ggsave(filename = "../output/20240302-ScPho4-CgPho4-align-region-graph.png",
       width = 5, height = 2)
```

## Add activation domain prediction result
Read data. Note the following instructions from the PADDLE ipython notebook script:

> Run predictions on all 53 amino acid long tiles across the protein.
> This function requires matching protein sequence and secondary structure scores.
> Returns a Numpy array of size (protein_length-52) which gives the
> predicted activation Z-score for the 53aa tiles starting at positions
> 1, 2, 3, ..., protein_length-52.
> High-strength ADs can be called by finding >=5 consecutive positions with Z-score > 6.
> Medium-strength ADs can be called by finding >=5 consecutive positions with Z-score > 4.

```{r}
paddle.Sc <- read_csv("../input/paddle-prediction/20240305-ScPho4-PADDLE-prediction.csv") %>% 
  mutate(
    strong = Z_score >= 6,
    medium = Z_score >= 4
  )

# extend the prediction score to the N-terminus and C-terminus
paddle.Sc <- paddle.Sc[c(rep(1, 26), 1:nrow(paddle.Sc), rep(nrow(paddle.Sc), 26)),] %>% 
  mutate(Pos = 1:n())
  

paddle.Cg <- read_csv("../input/paddle-prediction/20240305-CgPho4-PADDLE-prediction.csv") %>% 
  mutate(
    strong = Z_score >= 6,
    medium = Z_score >= 4
  )

# extend the prediction score to the N-terminus and C-terminus
paddle.Cg <- paddle.Cg[c(rep(1, 26), 1:nrow(paddle.Cg), rep(nrow(paddle.Cg), 26)),] %>% 
  mutate(Pos = 1:n())
```

```{r}
act.colors <- RColorBrewer::brewer.pal(n = 3, name = "Blues")
act.colors[1] = "gray50"
p1 <- p + 
  new_scale_fill() +
  geom_tile(data = paddle.Sc,
            aes(x = Pos - 312/2, y = 2.25, height = 0.05, fill = Z_score),
            width = 2, linewidth = 0) +
  geom_tile(data = paddle.Cg,
            aes(x = Pos - 533/2, y = 0.75, height = 0.05, fill = Z_score),
            width = 2, linewidth = 0) +
  scale_fill_stepsn("Activation", colors = act.colors,
                       breaks = c(-0.5, 4, 6, 10))

p1 <- p1 + theme_cowplot() +
  theme(legend.position = "bottom",
        legend.direction = "vertical",
        legend.box = "horizontal",
        legend.justification = "center",
        axis.line = element_blank(),
        axis.text = element_blank(),
        axis.title = element_blank(),
        axis.ticks = element_blank())

print(p1)
```

for the paper
```{r}
ggsave(filename = "../output/20240306-ScPho4-CgPho4-align-region-with-activation.png",
       width = 5, height = 3.5)
```

